sdks/java/core/src/main/java/org/apache/beam/sdk/util/PubsubClient.java

```diff
@@ -230,7 +232,7 @@ public String getPath() {
 
     public String getV1Beta1Path() {
       String[] splits = path.split("/");
-      Preconditions.checkState(splits.length == 4);
+      checkState(splits.length == 4);
```

Comment:
In general, would strongly prefer to have better error messages at all of these validation checks. For example,

```java
checkState(splits.length == 4, "Expected 4 path components in Pubsub path %s", path);
```